### PR TITLE
chore: switch to use management sdk

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

- Switch to use management SDK
- Remove the legacy-cloud-provider dependency for management operations.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
the unit test of setClient() func will be supplemented when fakeDriver is added.

**Release note**:
```

```

### CODE REVIEW

File: pkg/blobfuse/controllerserver.go [hunk 1]
```diff
@@ -144,28 +144,17 @@ func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest)
...
-	client, err := azstorage.NewBasicClientOnSovereignCloud(accountName, accountKey, d.cloud.Environment)
-	if err != nil {
-		return nil, err
-	}
-	blobClient := client.GetBlobService()
-	container := blobClient.GetContainerReference(containerName)
-	// todo: check what value to add into DeleteContainerOptions
 	err = wait.ExponentialBackoff(d.cloud.RequestBackoff(), func() (bool, error) {
-		_, err := container.DeleteIfExists(nil)
+		_, err = d.client.Delete(ctx, resourceGroupName, accountName, containerName)
```
Comment (andyzhangx) [inline pkg/blobfuse/controllerserver.go:152]: is it identical?

File: pkg/blobfuse/controllerserver.go [hunk 2]
```diff
@@ -193,24 +182,9 @@ func (d *Driver) ValidateVolumeCapabilities(ctx context.Context, req *csi.Valida
...
-
-	client, err := azstorage.NewBasicClientOnSovereignCloud(accountName, accountKey, d.cloud.Environment)
-	if err != nil {
-		return nil, err
-	}
-	blobClient := client.GetBlobService()
-	container := blobClient.GetContainerReference(containerName)
-
-	exist, err := container.Exists()
+	_, err = d.client.Get(ctx, resourceGroupName, accountName, containerName)
```
Comment (andyzhangx) [inline pkg/blobfuse/controllerserver.go:188]: is it identical to
```
exist, err := container.Exists()
```

File: pkg/blobfuse/controllerserver.go [hunk 3]
```diff
@@ -81,9 +81,9 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		resourceGroup = d.cloud.ResourceGroup
 	}
 
-	account, accountKey, err := d.cloud.EnsureStorageAccount(accountName, storageAccountType, string(storage.StorageV2), resourceGroup, location, blobfuseAccountNamePrefix)
+	account, _, err := d.cloud.EnsureStorageAccount(accountName, storageAccountType, string(storage.StorageV2), resourceGroup, location, blobfuseAccountNamePrefix)
```
Comment (andyzhangx) [inline pkg/blobfuse/controllerserver.go:84]: we still depends on `d.cloud.EnsureStorageAccount`, there is not much benefit for this PR.

File: pkg/blobfuse/controllerserver.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/blobfuse/controllerserver.go:152]: Currently, the effect is the same. Just replace the old SDK to the new one.

**Pull request closed without merging**